kernel/ukvm/kernel.c

```diff
@@ -27,12 +27,26 @@ void _start(struct ukvm_boot_info *bi)
     printf("\\__ \\ (   | | (   |  ) |\n");
     printf("____/\\___/ _|\\___/____/\n");
 
-    gdt_init();
+    /* It appears that on macosx Hypervisor.framework, the mxcsr is
+     * not started with its default value of 0x1f80.  This results in
+     * floating point related exceptions (e.g., denormal numbers)
+     * being raised to the guest which would normally be masked and
+     * dealt with by hardware (e.g., by rounding).  It is unclear
+     * whether this is a problem with Hypervisor.framework or the way
+     * that `uhvf` sets up the VMX context.
+     *
+     * A workaround is for the guest to explicitly set the MXCSR to
+     * the default value (0x1f80).
+     */
+    unsigned default_mxcsr = 0x1f80;
+    __asm__ __volatile__("ldmxcsr %0\n" : : "m"(default_mxcsr));
```

Comment:
The odd thing is that I can't find a similar workaround in xHyve for the mxcsr value... how does it deal with this idiosyncracy I wonder.